net-im/telegram-desktop-bin/telegram-desktop-bin-1.4.3-r1.ebuild

```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit desktop gnome2-utils xdg
+
+DESCRIPTION="Official desktop client for Telegram (binary package)"
+HOMEPAGE="https://desktop.telegram.org"
+SRC_URI="
+	https://github.com/telegramdesktop/tdesktop/archive/v${PV}.tar.gz -> tdesktop-${PV}.tar.gz
+	amd64? ( https://updates.tdesktop.com/tlinux/tsetup.${PV}.tar.xz )
+	x86? ( https://updates.tdesktop.com/tlinux32/tsetup32.${PV}.tar.xz )
+"
+
+LICENSE="telegram"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+
+QA_PREBUILT="usr/lib/${PN}/Telegram"
+
+RDEPEND="
+	dev-libs/glib:2
+	dev-libs/gobject-introspection
+	>=sys-apps/dbus-1.4.20
+	x11-libs/libX11
+	>=x11-libs/libxcb-1.10[xkb]
+	>=media-libs/fontconfig-2.13
+"
+
+S="${WORKDIR}/Telegram"
+
+src_prepare() {
+	sed -i -e 's/^Icon=telegram$/Icon=telegram-desktop/' "${WORKDIR}/tdesktop-${PV}"/lib/xdg/telegramdesktop.desktop || die
```

Comment:
I guess I'm being weird and all but *why* are you altering upstream .desktop file instead of fixing the install phase below to install the icon with correct name?